src/ardrive.ts

```diff
@@ -1521,4 +1532,40 @@ export class ArDrive extends ArDriveAnonymous {
 	async assertValidPassword(password: string): Promise<void> {
 		await this.arFsDao.assertValidPassword(password);
 	}
+
+	/**
+	 *
+	 * @param folderId - the ID of the folder to be download
+	 * @returns - the array of streams to write
+	 */
+	async downloadPrivateFolder(folderId: FolderID, maxDepth: number, path: string, driveKey: DriveKey): Promise<void> {
+		const folderEntityDump = await this.listPrivateFolder({ folderId, maxDepth, includeRoot: true, driveKey });
+		const rootFolder = folderEntityDump[0];
+		const rootFolderPath = rootFolder.path;
+		const basePath = rootFolderPath.replace(/\/[^/]+$/, '');
```

Comment:
Clarify this

Reply:
This code was de-coupled from the CLI layer https://github.com/ardriveapp/ardrive-core-js/pull/83